sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolDao.java

```diff
@@ -143,40 +160,104 @@ public Session currentSession() {
+    Criteria criteria = createCriteria();
+    criteria.add(Restrictions.in("identificationBarcode", barcodeList));
+    @SuppressWarnings("unchecked")
+    List<Pool> results = criteria.list();
+    return withWatcherGroup(results);
+  }
+
+  public JdbcTemplate getJdbcTemplate() {
+    return jdbcTemplate;
+  }
+
+  @Override
   public Pool getPoolByBarcode(String barcode, PlatformType platformType) throws IOException {
     if (barcode == null) throw new NullPointerException("cannot look up null barcode");
     if (platformType == null) {
       return getByBarcode(barcode);
     }
-    List<Pool> pools = listAllByCriteria(platformType, null, null, false);
+    List<Pool> pools = listAllByCriteria(platformType, barcode, null, false);
     return pools.size() == 1 ? pools.get(0) : null;
   }
 
+  @Override
+  public Map<String, Integer> getPoolColumnSizes() throws IOException {
+    return DbUtils.getColumnSizes(jdbcTemplate, TABLE_NAME);
+  }
+
+  private Group getPoolWatcherGroup() throws IOException {
+    return securityStore.getGroupByName("PoolWatchers");
+  }
+
+  public SecurityStore getSecurityStore() {
+    return securityStore;
+  }
+
   public SessionFactory getSessionFactory() {
     return sessionFactory;
   }
 
+  public WatchManager getWatchManager() {
+    return watchManager;
+  }
+
   @Override
   public List<Pool> listAll() throws IOException {
     @SuppressWarnings("unchecked")
     List<Pool> results = createCriteria().list();
     return withWatcherGroup(results);
   }
 
-  private List<Pool> withWatcherGroup(List<Pool> pools) throws IOException {
-    Group group = getPoolWatcherGroup();
-    for (Pool pool : pools) {
-      pool.setWatchGroup(group);
+  @Override
+  public List<Pool> listAllByCriteria(PlatformType platformType, String query, Integer limit, boolean ready) throws IOException {
+    if (limit != null && limit == 0) {
+      return Collections.emptyList();
     }
-    return pools;
+    Criteria criteria = createCriteria();
+    if (platformType != null) {
+      criteria.add(Restrictions.eq("platformType", platformType));
+    }
+    if (query != null) {
+      criteria.add(DbUtils.searchRestrictions(query, SEARCH_PROPERTIES));
+    }
+    if (limit != null) {
+      criteria.setMaxResults(limit);
+    }
+    if (ready) {
+      criteria.add(Restrictions.eq("readyToRun", true));
+    }
+    @SuppressWarnings("unchecked")
+    List<Pool> results = criteria.list();
+    return withWatcherGroup(results);
   }
 
   @Override
   public List<Pool> listByLibraryId(long libraryId) throws IOException {
     Criteria idCriteria = currentSession().createCriteria(LibraryImpl.class);
     idCriteria.add(Restrictions.eq("id", libraryId));
-    idCriteria.createAlias("libraryDilutions.pools.id", "poolIds");
-    idCriteria.setProjection(Projections.distinct(Projections.property("poolIds")));
+    idCriteria.createAlias("libraryDilutions", "libraryDilutions");
```

Comment:
I'm pretty sure this line (where 1 = 1) ends up being redundant for things that are not `derivedInfo`... (same below)